pkg/sdn/plugin/registry.go

```diff
@@ -287,14 +290,26 @@ func (registry *Registry) RunEventQueue(resourceName ResourceName) *oscache.Even
 		log.Fatalf("Unknown resource %s during initialization of event queue", resourceName)
 	}
 
-	lw := cache.NewListWatchFromClient(client, strings.ToLower(string(resourceName)), kapi.NamespaceAll, fields.Everything())
+	lw := cache.NewListWatchFromClient(client, strings.ToLower(string(resourceName)), kapi.NamespaceAll, fields)
 	eventQueue := oscache.NewEventQueue(cache.MetaNamespaceKeyFunc)
 	// Repopulate event queue every 30 mins
 	// Existing items in the event queue will have watch.Modified event type
 	cache.NewReflector(lw, expectedType, eventQueue, 30*time.Minute).Run()
 	return eventQueue
 }
 
+func (registry *Registry) RunLocalPodsEventQueue(nodeName string) *oscache.EventQueue {
+	if nodeName == "" {
+		log.Fatalf("LocalPods resource requires a node name")
+	}
+	lw := cache.NewListWatchFromClient(registry.kClient, strings.ToLower(string(Pods)), kapi.NamespaceAll, fields.Set{"spec.host": nodeName}.AsSelector())
+	eventQueue := oscache.NewEventQueue(cache.MetaNamespaceKeyFunc)
```

Comment:
Yes, we don't want to add new uses of anything in `pkg/client/cache` in Origin because we have moved almost everything they did upstream now (DelayingWorkqueue is the retry controller, DeltaFIFO is EventQueue).